src/main/scala/mesosphere/marathon/core/storage/store/impl/memory/InMemoryPersistenceStore.scala

```diff
@@ -148,6 +148,26 @@ class InMemoryPersistenceStore(implicit
 
     Future.successful(Done)
   }
+
+  @volatile private[this] var migrationInProgress: Boolean = false
+
+  override def startMigration(): Future[Done] = {
+    require(isOpen, "the store must be opened before it can be used")
+    if (migrationInProgress) {
+      throw new IllegalStateException("Migration is already in progress")
```

Comment:
Does it make sense to return`Future.failed` here?

Reply:
Will do.